dbms/src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2103,6 +2098,40 @@ bool StorageReplicatedMergeTree::fetchPart(const String & part_name, const Strin
 		MergeTreeData::Transaction transaction;
 		auto removed_parts = data.renameTempPartAndReplace(part, nullptr, &transaction);
 
+		std::shared_ptr<PartLog> part_log = context.getPartLog();
+		if (part_log)
```

Comment:
BTW, you could write 
`if (std::shared_ptr<PartLog> part_log = context.getPartLog())`
and avoid
`part_log.reset();`